silkworm/silkrpc/commands/ots_api.cpp

```diff
@@ -77,4 +79,105 @@ boost::asio::awaitable<void> OtsRpcApi::handle_ots_has_code(const nlohmann::json
+
+        const auto block_body_rlp = co_await core::rawdb::read_body_rlp(tx_database, block_hash, block_number);
+
+        if (block_body_rlp.empty()) {
+            throw std::runtime_error{"empty block body RLP in read_body"};
+        }
+
+        silkworm::ByteView data_view{block_body_rlp};
+        auto stored_body{silkworm::db::detail::decode_stored_block_body(data_view)};
+
+        const BlockDetails block_details{block_hash, block_header, total_difficulty, stored_body.txn_count -2, stored_body.ommers};
+
+        const BlockDetailsResponse block_details_response{block_details};
+
+        reply = make_json_content(request["id"], block_details_response);
+    } catch (const std::invalid_argument& iv) {
+        SILKRPC_WARN << "invalid_argument: " << iv.what() << " processing request: " << request.dump() << "\n";
+        reply = make_json_content(request["id"], nlohmann::detail::value_t::null);
+    } catch (const std::exception& e) {
+        SILKRPC_ERROR << "exception: " << e.what() << " processing request: " << request.dump() << "\n";
+        reply = make_json_error(request["id"], 100, e.what());
+    } catch (...) {
+        SILKRPC_ERROR << "unexpected exception processing request: " << request.dump() << "\n";
+        reply = make_json_error(request["id"], 100, "unexpected exception");
+    }
+
+    co_await tx->close();
+    co_return;
+}
+
+boost::asio::awaitable<void> OtsRpcApi::handle_ots_getBlockDetailsByHash(const nlohmann::json& request, nlohmann::json& reply) {
+    auto params = request["params"];
+    if (params.size() != 1) {
+        auto error_msg = "invalid ots_getBlockDetailsByHash params: " + params.dump();
+        SILKRPC_ERROR << error_msg << "\n";
+        reply = make_json_error(request["id"], 100, error_msg);
+        co_return;
+    }
+    auto block_hash = params[0].get<evmc::bytes32>();
+
+    SILKRPC_DEBUG << "block_hash: " << block_hash << "\n";
+
+    auto tx = co_await database_->begin();
+
+    try {
+        ethdb::TransactionDatabase tx_database{*tx};
+
+        const auto block_number = co_await read_header_number(tx_database, block_hash);
+        const auto block_header = co_await core::rawdb::read_header(tx_database, block_hash, block_number);
+        const auto total_difficulty = co_await core::rawdb::read_total_difficulty(tx_database, block_hash, block_number);
+
+        const auto block_body_rlp = co_await core::rawdb::read_body_rlp(tx_database, block_hash, block_number);
+
+        if (block_body_rlp.empty()) {
+            throw std::runtime_error{"empty block body RLP in read_body"};
+        }
+
+        silkworm::ByteView data_view{block_body_rlp};
+        auto stored_body{silkworm::db::detail::decode_stored_block_body(data_view)};
+
+        const BlockDetails block_details{block_hash, block_header, total_difficulty, stored_body.txn_count -2, stored_body.ommers};
+
+        const BlockDetailsResponse block_details_response{block_details};
```

Comment:
Two fields in `BlockDetailsResponse` are missing here:

- `issuance`: it should filled using `ethash::compute_reward`
- `total_fees`: it should be filled using `core::get_receipts`

However, there is more work to do on these subjects:
- regarding the first item, we should more generally add `compute_reward` method to `consensus::Engine` class hierarchy and let each consensus algorithm implement its own block reward
- regarding the second one, we should complete the implementation of `core::get_receipts` for the case when receipts are not present on the data store and must be recomputed by executing the call

These evolutions are not required for this PR, you can do them on next PR.